src/sentry/models/release.py

```diff
@@ -258,13 +258,10 @@ def set_refs(self, refs, user, fetch=False):
         ).exclude(version=self.version).order_by('-date_added').first()
 
         for ref in refs:
-            try:
-                repo = Repository.objects.get(
-                    organization_id=self.organization_id,
-                    name=ref['repository'],
-                )
-            except Repository.DoesNotExist:
-                continue
+            repo = Repository.objects.get(
+                organization_id=self.organization_id,
+                name=ref['repository'],
+            )
```

Comment:
ok, kinda thought of that, but will need to raise here so i can handle in the endpoint and send the appropriate response. should i create a `InvalidRepoName` exception? or just raise a `Repository.DoesNotExist`?